File: avalon-support/avalon-oauth2-support/avalon-oauth2-common/src/test/java/io/github/lishangbu/avalon/oauth2/common/properties/Oauth2PropertiesTest.java
package io.github.lishangbu.avalon.oauth2.common.properties;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertIterableEquals;
import static org.junit.jupiter.api.Assertions.assertNull;

import java.time.Duration;
import java.util.List;
import org.junit.jupiter.api.Test;

class Oauth2PropertiesTest {

    @Test
    void defaultsArePopulatedAndDurationParsed() {
        Oauth2Properties properties = new Oauth2Properties();

        assertIterableEquals(List.of(), properties.getIgnoreUrls());
        assertEquals("username", properties.getUsernameParameterName());
        assertEquals("password", properties.getPasswordParameterName());
        assertEquals(5, properties.getMaxLoginFailures());
        assertEquals(Duration.ofMinutes(15), properties.getLoginLockDuration());
    }

    @Test
    void durationParsingSupportsNumericString() {
        Oauth2Properties properties = new Oauth2Properties();

        properties.setLoginLockDuration("30");

        assertEquals(Duration.ofMinutes(30), properties.getLoginLockDuration());
    }

    @Test
    void durationParsingReturnsNullForBlankOrInvalid() {
        Oauth2Properties properties = new Oauth2Properties();
        properties.setLoginLockDuration("   ");
        assertNull(properties.getLoginLockDuration());

        properties.setLoginLockDuration("not-a-duration");
        assertNull(properties.getLoginLockDuration());

        properties.setLoginLockDuration(null);
        assertNull(properties.getLoginLockDuration());
    }

    @Test
    void settersUpdateValues() {
        Oauth2Properties properties = new Oauth2Properties();

        properties.setIgnoreUrls(List.of("/health", "/docs"));
        properties.setUsernameParameterName("user");
        properties.setPasswordParameterName("pwd");
        properties.setMaxLoginFailures(10);
        properties.setIssuerUrl("https://issuer");
        properties.setJwtPublicKeyLocation("classpath:pub.pem");
        properties.setJwtPrivateKeyLocation("classpath:pri.pem");

        assertIterableEquals(List.of("/health", "/docs"), properties.getIgnoreUrls());
        assertEquals("user", properties.getUsernameParameterName());
        assertEquals("pwd", properties.getPasswordParameterName());
        assertEquals(10, properties.getMaxLoginFailures());
        assertEquals("https://issuer", properties.getIssuerUrl());
        assertEquals("classpath:pub.pem", properties.getJwtPublicKeyLocation());
        assertEquals("classpath:pri.pem", properties.getJwtPrivateKeyLocation());
    }
}

File: avalon-support/avalon-oauth2-support/avalon-oauth2-common/src/test/kotlin/io/github/lishangbu/avalon/oauth2/common/properties/Oauth2PropertiesTest.kt
package io.github.lishangbu.avalon.oauth2.common.properties

import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Test
import java.time.Duration

class Oauth2PropertiesTest {
    @Test
    fun defaultsArePopulatedAndDurationParsed() {
        val properties = Oauth2Properties()

        assertIterableEquals(emptyList<String>(), properties.ignoreUrls)
        assertEquals("username", properties.usernameParameterName)
        assertEquals("password", properties.passwordParameterName)
        assertEquals(5, properties.maxLoginFailures)
        assertEquals(Duration.ofMinutes(15), properties.getLoginLockDuration())
    }

    @Test
    fun durationParsingSupportsNumericString() {
        val properties = Oauth2Properties()

        properties.setLoginLockDuration("30")

        assertEquals(Duration.ofMinutes(30), properties.getLoginLockDuration())
    }

    @Test
    fun durationParsingReturnsNullForBlankOrInvalid() {
        val properties = Oauth2Properties()
        properties.setLoginLockDuration("   ")
        assertNull(properties.getLoginLockDuration())

        properties.setLoginLockDuration("not-a-duration")
        assertNull(properties.getLoginLockDuration())

        properties.setLoginLockDuration(null)
        assertNull(properties.getLoginLockDuration())
    }

    @Test
    fun settersUpdateValues() {
        val properties = Oauth2Properties()

        properties.ignoreUrls = mutableListOf("/health", "/docs")
        properties.usernameParameterName = "user"
        properties.passwordParameterName = "pwd"
        properties.maxLoginFailures = 10
        properties.issuerUrl = "https://issuer"
        properties.jwtPublicKeyLocation = "classpath:pub.pem"
        properties.jwtPrivateKeyLocation = "classpath:pri.pem"

        assertIterableEquals(listOf("/health", "/docs"), properties.ignoreUrls)
        assertEquals("user", properties.usernameParameterName)
        assertEquals("pwd", properties.passwordParameterName)
        assertEquals(10, properties.maxLoginFailures)
        assertEquals("https://issuer", properties.issuerUrl)
        assertEquals("classpath:pub.pem", properties.jwtPublicKeyLocation)
        assertEquals("classpath:pri.pem", properties.jwtPrivateKeyLocation)
    }
}
